src/pymatgen/io/lobster/outputs.py

```diff
@@ -1640,12 +1738,52 @@ def __init__(
             small_dict: dict[str, Any] = {}
             for line in lines[3:]:
                 cleanlines = [idx for idx in line.split(" ") if idx != ""]
-                if len(cleanlines) == 5:
+                if len(cleanlines) == 5 and cleanlines[0].isdigit() and "LCFO" not in Path(self._filename).name:
                     small_dict = {"Mulliken GP": {}, "Loewdin GP": {}, "element": cleanlines[1]}
                     small_dict["Mulliken GP"][cleanlines[2]] = float(cleanlines[3])
                     small_dict["Loewdin GP"][cleanlines[2]] = float(cleanlines[4])
+                elif len(cleanlines) == 4 and cleanlines[0].isdigit() and "LCFO" in Path(self._filename).name:
+                    small_dict = {"Loewdin GP": {}, "mol": cleanlines[1]}
+                    small_dict["Loewdin GP"][cleanlines[2]] = float(cleanlines[3])
+                elif len(cleanlines) == 5 and cleanlines[0].isdigit() and "LCFO" in Path(self._filename).name:
+                    small_dict = {"Loewdin GP": {}, "mol": cleanlines[1]}
+                    small_dict["Loewdin GP"][cleanlines[2]] = {
+                        Spin.up: float(cleanlines[3]),
+                        Spin.down: float(cleanlines[4]),
+                    }
+                elif len(cleanlines) == 5 and not cleanlines[0].isdigit():
+                    small_dict["Mulliken GP"][cleanlines[0]] = {
+                        Spin.up: float(cleanlines[1]),
+                        Spin.down: float(cleanlines[2]),
+                    }
+                    small_dict["Loewdin GP"][cleanlines[0]] = {
+                        Spin.up: float(cleanlines[3]),
+                        Spin.down: float(cleanlines[4]),
+                    }
+                    if "total" in cleanlines[0]:
+                        self.list_dict_grosspop.append(small_dict)
+                elif len(cleanlines) == 7 and cleanlines[0].isdigit():
+                    small_dict = {"Mulliken GP": {}, "Loewdin GP": {}, "element": cleanlines[1]}
+                    small_dict["Mulliken GP"][cleanlines[2]] = {
+                        Spin.up: float(cleanlines[3]),
+                        Spin.down: float(cleanlines[4]),
+                    }
+                    small_dict["Loewdin GP"][cleanlines[2]] = {
+                        Spin.up: float(cleanlines[5]),
+                        Spin.down: float(cleanlines[6]),
+                    }
 
-                elif len(cleanlines) > 0:
+                elif len(cleanlines) > 0 and "spin" not in line and "LCFO" in Path(self._filename).name:
```

Comment:
Is there another way to check this? Not sure that the filename alone is safe enough

Reply:
Thanks, updated the code to check if it is atom or molecule type from the file data itself. 

However, I still rely on filename checks in Grosspop.get_structure_with_total_grosspop / Charge.get_structure_with_charges methods. If I want to do this check using file data, I would need to define another argument to the function and set an attribute, `_file_type` to the class to keep it MSONable.